### PR TITLE
Add Java text sample.

### PR DESCRIPTION
**Pull request by tswast**
I don't include as thorough as documentation as the Python sample, but perhaps we can merge the two if/when we write a docs site page?

### CODE REVIEW

**Comment by amygdala** [comment]
oh, nice.  Will look by tomorrow.  Yes, let's plan to merge when we port to docs page.

File: java/text/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+Download Tokenizer data and save it to this directory.
+
+    wget http://opennlp.sourceforge.net/models-1.5/en-token.bin
+
+## Run the sample
+
+To build and run the sample, run the following from this directory:
+
+    mvn clean compile assembly:single
+    java -cp target/text-1.0-SNAPSHOT-jar-with-dependencies.jar com.google.cloud.vision.samples.text.TextApp ../../data/text/wakeupcat.jpg
```
Comment (amygdala) [inline java/text/README.md]: Maybe mention here that you can pass it a directory path arg as well.

**Comment by amygdala** [comment]
This is really nice.  Haven't finished, but a couple of initial general comments:
- Apparently, this version partially but not entirely shares redis structure with the python version -- the python version knows that a file indexed by the java version is a 'hit', but can't extract the text. Would be nice to have them consistent so that the 2 programs can share the same redis db.  (I'm about to poke into this further).
- It doesn't look like it keeps track of what it has indexed (in redis), in the sense that if you pass it a directory, interrupt it, and start it up again, it starts over.  That might be nice to add, maybe in a later phase.

File: java/text/README.md (hunk 1 above)
Comment (tswast) [inline java/text/README.md]: Done. I actually changed this to just the directory instructions. I wrote the README before I figured out how to walk a directory in Java. The fact that it works with just a single file was unintended.

**Comment by tswast** [comment]
The DBs are now compatible. Also, I fixed the resume issue. The problem was that I was extracting the text from all images before adding them to the DB. It's better and simpler this way. Thanks for reviewing.

**Comment by amygdala** [comment]
LGTM.  Fun :).  Do you want to maybe squash in the branch before we pull in?

**Comment by tswast** [comment]
Squashed.

**Pull request merged**